src/main/java/uk/nhs/hee/tis/trainee/sync/facade/ProgrammeMembershipEnricherFacade.java

```diff
@@ -95,6 +95,34 @@
     this.tcsSyncService = tcsSyncService;
   }
 
+  public Optional<ProgrammeMembership> getProgrammeMembershipById(String id) {
+    return programmeMembershipService.findById(id);
+  }
+
+  /**
+   * Delete a programmeMembership from tis-trainee-details.
+   *
+   * @param programmeMembership The programme membership to delete.
+   */
+  public void delete(ProgrammeMembership programmeMembership) {
```

Comment:
Not sure if I understand the logic behind this method. Is it deleting all programmeMemberships of a person and resynching the remaining ones that haven't been deleted?

Reply:
Yes, that's right. The reason is that a single programmeMembership record is not necessarily passed to trainee-details as-such, it is first aggregated to collect the curriculum details from the set of related programmeMemberships. So if one PM is deleted, you need to rebuild the set to which it belongs.